src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/VisualStudioMacLSPEditorFeatureDetector.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.Editor.Razor;
+using MonoDevelop.Core.FeatureConfiguration;
+using MonoDevelop.Ide;
+using MonoDevelop.Projects;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Shared]
+    [Export(typeof(LSPEditorFeatureDetector))]
+    internal class VisualStudioMacLSPEditorFeatureDetector : LSPEditorFeatureDetector
+    {
+        private const string RazorLSPEditorFeatureFlag = "Razor.LSP.Editor";
+        private const string DotNetCoreCSharpCapability = "CSharp&CPS";
+        private const string LegacyRazorEditorCapability = "LegacyRazorEditor";
+
+        private readonly AggregateProjectCapabilityResolver _projectCapabilityResolver;
+        private readonly TextBufferProjectService _textBufferProjectService;
+        private readonly Lazy<bool> _useLegacyEditor;
+
+        [ImportingConstructor]
+        public VisualStudioMacLSPEditorFeatureDetector(
+            AggregateProjectCapabilityResolver projectCapabilityResolver,
+            TextBufferProjectService textBufferProjectService)
+        {
+            _projectCapabilityResolver = projectCapabilityResolver;
+            _textBufferProjectService = textBufferProjectService;
+
+            _useLegacyEditor = new Lazy<bool>(() =>
+            {
+                // TODO: Pull from preview features collection
```

Comment:
Not allowed to do this today because adding anything to the preview features collection comes with UI which is gated.

Reply:
/cc @KirillOsenkov @gundermanc

Reply:
Well, it’s not that bad. We can add the checkbox to VSMac, but it needs to be working somewhat.